src/deploy/index.js

```diff
@@ -38,22 +41,21 @@ const deploySite = async (api, siteId, dir, opts) => {
       //     spinner: a spinner from cli-spinners package
       // }
     },
-    // allows updating an existing deploy
-    deployId: null,
-    ...opts,
-  }
-
-  const { fnDir, configPath, statusCb, message: title } = opts
-
+    deployId: deployIdOpt = null,
```

Comment:
Distinguish between `opts.deploy` (passed by user) and `deploy.id` (returned by API).